azurerm/resource_arm_container_group.go

```diff
@@ -0,0 +1,323 @@
+	name := d.Get("name").(string)
+	location := d.Get("location").(string)
+	OSType := d.Get("os_type").(string)
+	IPAddressType := d.Get("ip_address_type").(string)
+	tags := d.Get("tags").(map[string]interface{})
+
+	containers, containerGroupPorts := expandContainerGroupContainers(d)
+
+	containerGroup := containerinstance.ContainerGroup{
+		Name:     &name,
+		Location: &location,
+		Tags:     expandTags(tags),
+		ContainerGroupProperties: &containerinstance.ContainerGroupProperties{
+			Containers: containers,
+			IPAddress: &containerinstance.IPAddress{
+				Type:  &IPAddressType,
+				Ports: containerGroupPorts,
+			},
+			OsType: containerinstance.OperatingSystemTypes(OSType),
+		},
+	}
+
+	_, err := containerGroupsClient.CreateOrUpdate(resGroup, name, containerGroup)
+	if err != nil {
+		return err
+	}
+
+	read, err := containerGroupsClient.Get(resGroup, name)
+	if err != nil {
+		return err
+	}
+
+	if read.ID == nil {
+		return fmt.Errorf("Cannot read container group %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*read.ID)
+
+	return resourceArmContainerGroupRead(d, meta)
+}
+func resourceArmContainerGroupRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	resp, err := containterGroupsClient.Get(resGroup, name)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("location", azureRMNormalizeLocation(*resp.Location))
+	flattenAndSetTags(d, resp.Tags)
+
+	d.Set("os_type", string(resp.OsType))
+	if address := resp.IPAddress; address != nil {
+		d.Set("ip_address_type", address.Type)
+		d.Set("ip_address", address.IP)
+	}
+
+	containerConfigs := flattenContainerGroupContainers(resp.Containers)
+	d.Set("container", containerConfigs)
+
+	return nil
+}
+
+func resourceArmContainerGroupDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient)
+	containterGroupsClient := client.containerGroupsClient
+
+	id, err := parseAzureResourceID(d.Id())
+
+	if err != nil {
+		return err
+	}
+
+	// container group properties
+	resGroup := id.ResourceGroup
+	name := id.Path["containerGroups"]
+
+	resp, err := containterGroupsClient.Delete(resGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return nil
+		}
+		return err
+	}
+
+	return nil
+}
+
+func flattenContainerGroupContainers(containers *[]containerinstance.Container) []interface{} {
+
+	containerConfigs := make([]interface{}, 0, len(*containers))
+	for _, container := range *containers {
+		containerConfig := make(map[string]interface{})
+		containerConfig["name"] = *container.Name
+		containerConfig["image"] = *container.Image
+
+		resourceRequests := *(*container.Resources).Requests
```

Comment:
could we add an `if`check around this to avoid Swagger/API issues? i.e.

```
if resources := container.Resources; resources != nil {
  if resourceRequests := resources.Requests; resourceRequests != nil {
    # ...
  }
}
```

Reply:
The API will always return these values. Changing just in case.